### PR TITLE
Automatically generate collection deprecation changelog entries from metadata

### PR DESCRIPTION
**Pull request by felixfontein**
Use https://github.com/ansible-community/antsibull/pull/623 to automatically generate collection deprecation changelog entries from metadata.

### CODE REVIEW

**Comment by mariolenz** [comment]
I should say the output looks OK.

BTW do you want to merge this eventually or is this only meant as an example how it would look like?

**Comment by felixfontein** [comment]
> BTW do you want to merge this eventually or is this only meant as an example how it would look like?

I'd merge it together with https://github.com/ansible-community/antsibull/pull/623, so that re-generating the changelogs (for example when the next release happens) doesn't result in changes of older entries.

**Comment by felixfontein** [comment]
I updated the PR to the latest version of https://github.com/ansible-community/antsibull/pull/623 and this repo.

**Comment by felixfontein** [comment]
#459 will allow to also generate the removal changelog fragments automatically.

File: 10/CHANGELOG-v10.md [hunk 1]
```diff
@@ -1585,8 +1585,32 @@ If not mentioned explicitly\, the changes are reported in the combined changelog
...
-* The <code>openvswitch\.openvswitch</code> collection has been deprecated by the maintainers\. Since they\'ve also announced to not support ansible\-core 2\.18\, it will be removed from Ansible 11 if no one starts maintaining it again before Ansible 11\. See [the removal process for details on how this works](https\://docs\.ansible\.com/ansible/devel/community/collection\_contributors/collection\_package\_removal\.html\#canceling\-removal\-of\-an\-unmaintained\-collection) \([https\://forum\.ansible\.com/t/6245](https\://forum\.ansible\.com/t/6245)\)\.
+* The <code>frr\.frr</code> collection has been deprecated\.
+  It will be removed from Ansible 11 if no one starts maintaining it again before Ansible \{removal\.major\_version\}\.
```
Comment (mariolenz) [inline 10/CHANGELOG-v10.md:1589]: It looks like `{removal.major_version}` is substituted correctly with 11 the first but not the second time.

File: 10/CHANGELOG-v10.md [hunk 2]
```diff
@@ -1585,8 +1585,32 @@ If not mentioned explicitly\, the changes are reported in the combined changelog
...
+  See <em class="title-reference">Collections Removal Process for unmaintained collections \<https\://docs\.ansible\.com/ansible/devel/community/collection\_contributors/collection\_package\_removal\.html\#unmaintained\-collections\></em> for more <a href="#system-message-1"><span class="problematic">details\_\_</span></a> \([https\://forum\.ansible\.com/t/6243](https\://forum\.ansible\.com/t/6243)\)\.
+
+  ## Docutils System Messages
```
Comment (mariolenz) [inline 10/CHANGELOG-v10.md:1592]: This looks like something went wrong here. But I don't understand what :shrug:

File: 10/CHANGELOG-v10.md (hunk 1 above)
Comment (felixfontein) [reply 10/CHANGELOG-v10.md]: Thanks, that should be fixed now.

File: 10/CHANGELOG-v10.md (hunk 2 above)
Comment (felixfontein) [reply 10/CHANGELOG-v10.md:1592]: Same here. That was likely caused from the `{}` expression...

Comment (mariolenz) [reply 10/CHANGELOG-v10.md:1592]: @felixfontein It looks like there are still Docutils System Messages in the file.

Comment (felixfontein) [reply 10/CHANGELOG-v10.md]: I grepped on the wrong place... now they're really gone :D

File: 10/CHANGELOG-v10.md [hunk 3]
```diff
@@ -1585,8 +1585,12 @@ If not mentioned explicitly\, the changes are reported in the combined changelog
...
 ### Deprecated Features
 
-* The <code>frr\.frr</code> collection has been deprecated by the maintainers\. Since they\'ve also announced to not support ansible\-core 2\.18\, it will be removed from Ansible 11 if no one starts maintaining it again before Ansible 11\. See [the removal process for details on how this works](https\://docs\.ansible\.com/ansible/devel/community/collection\_contributors/collection\_package\_removal\.html\#canceling\-removal\-of\-an\-unmaintained\-collection) \([https\://forum\.ansible\.com/t/6243](https\://forum\.ansible\.com/t/6243)\)\.
-* The <code>openvswitch\.openvswitch</code> collection has been deprecated by the maintainers\. Since they\'ve also announced to not support ansible\-core 2\.18\, it will be removed from Ansible 11 if no one starts maintaining it again before Ansible 11\. See [the removal process for details on how this works](https\://docs\.ansible\.com/ansible/devel/community/collection\_contributors/collection\_package\_removal\.html\#canceling\-removal\-of\-an\-unmaintained\-collection) \([https\://forum\.ansible\.com/t/6245](https\://forum\.ansible\.com/t/6245)\)\.
+* The <code>frr\.frr</code> collection has been deprecated\.
```
Comment (mariolenz) [inline 10/CHANGELOG-v10.md:1588]: @felixfontein `\.` instead of just `.` at the end of a sentence looks weird.

Comment (felixfontein) [reply 10/CHANGELOG-v10.md:1588]: That renders as `.`. Also this is not related to this PR and has been there before.

Ref: https://spec.commonmark.org/0.31.2/#backslash-escapes

**Review by mariolenz — APPROVED** [review]
Quite a lot of changes so it's hard to be sure, but I should say this LGTM now.

**Comment by felixfontein** [comment]
I'd like to regenerate the changelog (and docsite) once more once #464 is merged, to see whether the markup formatting works.

**Comment by mariolenz** [comment]
Makes sense. Better test this and make make sure.

**Comment by felixfontein** [comment]
(With the latest changes from https://github.com/ansible-community/antsibull/pull/623 and with #464 merged, the generated changelogs will look exactly as they already do. All changes only affect entries in versions not yet released. See https://github.com/ansible-community/antsibull/pull/623#issuecomment-2394999285 for how the result looks like for 11.0.0a2.)

**Comment by felixfontein** [comment]
I'll keep this marked as draft until https://github.com/ansible-community/antsibull/pull/623 has been merged, then I will merge the latest version of this.

**Comment by felixfontein** [comment]
The CI failure is unrelated.

**Pull request merged**